apps/rewards/src/invite/rewardSender.service.ts

```diff
@@ -0,0 +1,156 @@
+          this.watchedInvites.delete(invite)
+        } else if (
+          Date.now() - invite.sentAt >
+          this.appCfg.transactionTimeoutMs
+        ) {
+          await this.inviteRewardRepository.update(invite.inviteId, {
+            state: invite.markAsDeadLetterIfFailed
+              ? RewardStatus.DeadLettered
+              : RewardStatus.Failed
+          })
+          this.watchedInvites.delete(invite)
+        }
+      })
+    )
+  }
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  async checkFailedInvites() {
+    await this.inviteRewardRepository.manager.transaction(
+      async (entityManager: EntityManager) => {
+        const repository = entityManager.getRepository<InviteReward>(
+          'invite_reward'
+        )
+        const invites = await repository
+          .createQueryBuilder()
+          .setLock('pessimistic_read')
+          .where({
+            state: In([RewardStatus.Submitted, RewardStatus.Failed]),
+            createdAt: Raw(
+              alias =>
+                `${alias} <= current_timestamp - (30 ||' minutes')::interval`
+            )
+          })
+          .getMany()
+        if (invites.length > 0) {
+          this.logger.log(`Found failed or submitted invites ${invites.length}`)
+        }
+        await Promise.all(
+          invites.map(async invite => {
+            if (
+              invite.state === RewardStatus.Submitted &&
+              (await this.wasTxSentSuccesfully(invite.rewardTxHash))
+            ) {
+              await repository.update(invite.id, {
+                state: RewardStatus.Completed
+              })
+            } else {
+              await this.sendInviteReward(invite, true, repository)
+            }
+          })
+        )
+      }
+    )
+  }
+
+  async getRewardInCelo() {
+    const rewardInCusdWei = this.appCfg.inviteRewardAmountInCusd * WEI_PER_UNIT
+    const exchange = await this.contractKit.contracts.getExchange()
+    const exchangeRate = await exchange.getGoldExchangeRate(
+      new BigNumber(rewardInCusdWei)
+    )
+    return exchangeRate.multipliedBy(rewardInCusdWei).toFixed(0)
+  }
+
+  async sendInviteReward(
+    invite: InviteReward,
+    isRetry: boolean = false,
+    repository: Repository<InviteReward> = this.inviteRewardRepository
+  ) {
+    const celoToken = await this.contractKit.contracts.getGoldToken()
+    const tx = await celoToken.transfer(
+      invite.inviter,
+      await this.getRewardInCelo()
+    )
+
+    const resp = await this.relayerProxyService.submitTransaction(
+      {
+        transaction: {
+          destination: tx.txo._parent.options.address,
+          value: '0',
+          data: tx.txo.encodeABI()
+        }
+      },
+      invite.id
+    )
+    if (resp.ok === false) {
+      this.logger.error(
+        `Error sending reward for invite ${invite.id}: ${resp.error}`
+      )
+      await repository.update(invite.id, {
+        state: isRetry ? RewardStatus.DeadLettered : RewardStatus.Failed
```

Comment:
Does this imply that we retry max one time and then dead-letter it?

Reply:
Maybe add a comment to make it explicit

Reply:
It does, yes. IIUC the relayer already does some retrying so this is mostly to try once more 30 minutes after the first set of tries. Do you think that's not enough?

Reply:
No, I think that's good. Just wanted to add a comment stating that's what's going on since it may be hard to figure it out on first glance.